### DISCUSSION TITLE
Autocomplete API Input authentication

### DISCUSSION THREAD

**Discussion by NickSettler** [directus/directus#11192] [Feature Requests]
It would be greatful, if there would be an option to set authentification when using autocomplete API interface for field. Many of the apis require auth to access their data. I suppose currently there's no solution in Directus for setting authentification token when using Autocomplete API

**Comment by hermesalvesbr** [comment]
Did you manage to get around this?

**Reply by NickSettler** [reply]
Still no :(

**Reply by hermesalvesbr** [reply]
I'm thinking make a custom endpoint and call it.

**Reply by rijkvanzanten** [reply]
Sounds like the way to go to me 👍🏻 The problem with doing it as an option on the interface is that the token is gonna be exposed to the client, which is a bit of a security no-no. When you do it as a custom endpoint, you're ensured that the access token isn't leaked at all.

**Reply by orestesgaolin** [reply]
Has anyone managed to do custom interface for where we could provide headers? Ideally I'd like to also have custom display.
